src/map.cpp

```diff
@@ -8175,22 +8169,25 @@ template<typename Functor>
     }
 }
 
-void map::scent_blockers( bool (&blocks_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
-                          bool (&reduces_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
+void map::scent_blockers( std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &blocks_scent,
+                          std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &reduces_scent,
```

Comment:
> This would make an additional coupling.

Ah, I was talking about the two array containing the booleans, not the actual scent data itself. Those represent how objects interact with scent and are therefor properties of the map objects and the map. Therefor they can be cached in the map.

The scent data should of course belong into the `scent_map` class.

I'll leave it as it is because it is going to be addressed by Coolthulhu in the linked PR.